model/TestCenterService.php

```diff
@@ -201,4 +204,58 @@ public function getTestCenter($testCenterUri)
     {
         return new core_kernel_classes_Resource($testCenterUri);
     }
+
+    /**
+     * @param core_kernel_classes_Resource $testCenter
+     * @param User $user
+     * @param core_kernel_classes_Resource $role
+     * @return bool
+     * @throws TestCenterException
+     */
+    public function assignUser(\core_kernel_classes_Resource $testCenter, User $user, \core_kernel_classes_Resource $role)
+    {
+        $userResource = $this->getResource($user->getIdentifier());
+        $userRoles = $userResource->getPropertyValues($this->getProperty(UserRdf::PROPERTY_ROLES));
```

Comment:
You could save here a database hit by calling `$user->getRoles()`. What do you think?